ingestion/src/metadata/ingestion/source/dashboard/qlikcloud/metadata.py

```diff
@@ -0,0 +1,340 @@
+        if datamodel.tableName and db_service_entity:
+            try:
+                if len(datamodel.connectorProperties.tableQualifiers) > 1:
+                    (
+                        database_name,
+                        schema_name,
+                    ) = datamodel.connectorProperties.tableQualifiers[-2:]
+                elif len(datamodel.connectorProperties.tableQualifiers) == 1:
+                    schema_name = datamodel.connectorProperties.tableQualifiers[-1]
+                    database_name = None
+                else:
+                    schema_name, database_name = None, None
+
+                table_fqn = fqn.build(
+                    self.metadata,
+                    entity_type=Table,
+                    service_name=db_service_entity.name.__root__,
+                    schema_name=schema_name,
+                    table_name=datamodel.tableName,
+                    database_name=database_name,
+                )
+                if table_fqn:
+                    return self.metadata.get_by_name(
+                        entity=Table,
+                        fqn=table_fqn,
+                    )
+            except Exception as exc:
+                logger.debug(traceback.format_exc())
+                logger.warning(f"Error occured while finding table fqn: {exc}")
+        return None
+
+    def yield_dashboard_lineage_details(
+        self,
+        dashboard_details: QlikApp,
+        db_service_name: Optional[str],
+    ) -> Iterable[Either[AddLineageRequest]]:
+        """Get lineage method"""
+        db_service_entity = self.metadata.get_by_name(
+            entity=DatabaseService, fqn=db_service_name
+        )
+        for datamodel in self.data_models or []:
+            try:
+                data_model_entity = self._get_datamodel(datamodel=datamodel)
+                if data_model_entity:
+                    om_table = self._get_database_table(
+                        db_service_entity, datamodel=datamodel
+                    )
+                    if om_table:
+                        yield self._get_add_lineage_request(
+                            to_entity=data_model_entity, from_entity=om_table
+                        )
+            except Exception as err:
+                yield Either(
+                    left=StackTraceError(
+                        name=f"{dashboard_details.qDocName} Lineage",
+                        error=(
+                            "Error to yield dashboard lineage details for DB "
+                            f"service name [{db_service_name}]: {err}"
+                        ),
+                        stackTrace=traceback.format_exc(),
+                    )
+                )
+
+    def _get_database_service(self, db_service_name: str):
+        return self.metadata.get_by_name(DatabaseService, db_service_name)
+
+    def _yield_lineage_from_query(self) -> Iterable[Either[AddLineageRequest]]:
+        pass
```

Comment:
why are we defining these methods? Might be missing something, but I don't see where they come from